File: java/ankit_kanyal/devnest_problems/phase_1/InsertingIntervals.java
// link- https://devsnest.in/algo-challenges/inserting-intervals

static int[][] solve(int[][] intervals, int n, int[] arr){

Arrays.sort(intervals,Comparator.comparingInt(x->x[0]));
int prevStart=intervals[0][0];
int prevEnd=intervals[0][1];
int startCount=1;
if(arr[0]<prevStart){
    prevStart=arr[0];
    prevEnd=arr[1];
    startCount=0;
}
ArrayList<ArrayList<Integer>> list=new ArrayList<>();
ArrayList<Integer> temp;

boolean isAdded=false;
for(int i=startCount;i<intervals.length;i++){
    int curStart=intervals[i][0];
    int curEnd=intervals[i][1];

    if(!isAdded && arr[0]<curStart && prevEnd<arr[1]){
        if(prevEnd<arr[0]){
            temp=new ArrayList<>();
          temp.add(prevStart);
          temp.add(prevEnd);
          prevStart=arr[0];
          prevEnd=arr[1];
          list.add(temp);
        }
        else{
            prevEnd=arr[1];
        }
        isAdded=true;
    }

    if(prevEnd<curEnd){
        if(prevEnd<curStart){
            temp=new ArrayList<>();
          temp.add(prevStart);
          temp.add(prevEnd);
          prevStart=curStart;
          prevEnd=curEnd;
          list.add(temp);
        }
        else{
            prevEnd=curEnd;
        }
    }

}

temp=new ArrayList<>();
temp.add(prevStart);
temp.add(prevEnd);
list.add(temp);

int [][]ans=list.stream().map(l->l.stream().mapToInt(Integer::intValue).toArray()).toArray(int [][]::new);

return ans;
}

File: Python/vedant_bhatnagar/DevsnestProblems_THA/Phase 1/insertingIntervals.py
def solve(intervals, n, arr):
    # CODE HERE
    n = len(intervals)
    for i in range(0,n-1):
        if arr[0]>intervals[i][0] and arr[0]<intervals[i+1][0]:
            intervals = intervals[0:i+1] + [arr] + intervals[i+1:]
            break
        elif arr[0]<intervals[i][0]:
            intervals = [arr]+intervals
    # print(intervals)
    ans = []
    curr_interval = intervals[0]
    

    for next_interval in intervals[1:]:
        if curr_interval[1] >= next_interval[0]:
            if curr_interval[1] < next_interval[1]:
                curr_interval[1] = next_interval[1]
        else:
            ans.append(curr_interval)
            curr_interval = next_interval
    ans.append(curr_interval)

    return ans